autobahn/src/main/java/io/crossbar/autobahn/wamp/messages/Event.java

```diff
@@ -62,8 +71,14 @@ public static Event parse(List<Object> wmsg) {
         marshaled.add(MESSAGE_TYPE);
         marshaled.add(subscription);
         marshaled.add(publication);
-        // Empty details.
-        marshaled.add(Collections.emptyMap());
+        Map<String, Object> details = new HashMap<>();
+        if (topic != null) {
+			details.put("topic", topic);
+		}
+        if (retained) {
+        	details.put("retained", retained);
```

Comment:
4 spaces here as well.

Reply:
I'm a bit busy right now but I will change the indentation and the EventDetails class later today.